Providers/Scripts/3.x/Scripts/nxOMSAgentNPMConfig.py

```diff
@@ -0,0 +1,209 @@
+nxDSCLog = imp.load_source('nxDSCLog', '../nxDSCLog.py')
+LG = nxDSCLog.DSCLog
+
+global show_mof
+show_mof = False
+CONFIG_PATH = '/etc/opt/microsoft/omsagent/conf/'
+SERVER_ADDRESS = '/var/opt/microsoft/omsagent/run/npmdagent.sock'
+DEST_FILE_NAME = 'npmd_agent_config.conf'
+
+#   [key] string ConfigType;
+#   [write] string ConfigID;
+#   [write] string Contents;
+#   [write,ValueMap{"Present", "Absent"},Values{"Present", "Absent"}] string Ensure;
+#   [write] string ContentChecksum;
+
+def init_vars(ConfigType, ConfigID, Contents, Ensure, ContentChecksum):
+    if ConfigType is not None and ConfigType != '':
+        ConfigType = ConfigType.encode('ascii', 'ignore')
+    else:
+        ConfigType = 'UpdateNPMAgentConfig'
+    if ConfigID is not None:
+        ConfigID = ConfigID.encode('ascii', 'ignore')
+    else:
+        ConfigID = ''
+    if Contents is not None:
+        Contents = base64.b64decode(Contents)#Contents.encode('ascii', 'ignore')
+    else:
+        Contents = ''
+    if Ensure is not None and Ensure != '':
+        Ensure = Ensure.encode('ascii', 'ignore')
+    else:
+        Ensure = 'Present'
+    if ContentChecksum is not None:
+        ContentChecksum = ContentChecksum.encode('ascii', 'ignore')
+    else:
+        ContentChecksum = ''
+    return ConfigType, ConfigID, Contents, Ensure, ContentChecksum
+
+
+def Set_Marshall(ConfigType, ConfigID, Contents, Ensure, ContentChecksum):
+    (ConfigType, ConfigID, Contents, Ensure, ContentChecksum) = init_vars(ConfigType, ConfigID, Contents, Ensure, ContentChecksum)
+    retval = Set(ConfigType, ConfigID, Contents, Ensure, ContentChecksum)
+    return retval
+
+
+def Test_Marshall(ConfigType, ConfigID, Contents, Ensure, ContentChecksum):
+    (ConfigType, ConfigID, Contents, Ensure, ContentChecksum) = init_vars(ConfigType, ConfigID, Contents, Ensure, ContentChecksum)
+    retval = Test(ConfigType, ConfigID, Contents, Ensure, ContentChecksum)
+    return retval
+
+
+def Get_Marshall(ConfigType, ConfigID, Contents, Ensure, ContentChecksum):
+    arg_names = list(locals().keys())
+    (ConfigType, ConfigID, Contents, Ensure, ContentChecksum) = init_vars(ConfigType, ConfigID, Contents, Ensure, ContentChecksum)
+    retval = 0
+    retval = Get(ConfigType, ConfigID, Contents, Ensure, ContentChecksum)
+    ConfigType = protocol.MI_String(ConfigType)
+    ConfigID = protocol.MI_String(ConfigID)
+    Ensure = protocol.MI_String(Ensure)
+    Contents = protocol.MI_String(Contents)
+    ContentChecksum = protocol.MI_String(ContentChecksum)
+
+    retd = {}
+    ld = locals()
+    for k in arg_names:
+        retd[k] = ld[k]
+    return retval, retd
+
+
+############################################################
+# Begin user defined DSC functions
+############################################################
+
+def SetShowMof(a):
+    global show_mof
+    show_mof = a
+
+
+def ShowMof(op, ConfigType, ConfigID, Contents, Ensure, ContentChecksum):
```

Comment:
Since this method is creating or adding to the mof, please consider it renaming to reflect that